provider/filepath.go

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type Filepath interface {
+	Bucket() string
+	Prefix() string
+	Path() string
+	FullPathWithBucket() string
```

Comment:
Whats the difference between this and Path?

Reply:
This includes `s3://bucket/prefix/path` where path is `path`

Reply:
This seems like it's a filestorepath vs filepath right?

Reply:
my thinking is, if someone looks at this for the first time the question they're most likely going to ask is what's the diff b/w this and go's filepath?

Reply:
That's fair; so we do have different types like `S3Filepath` and will add additional paths as needed. But I can definitely modify the interface to FilestorePath if we want.